crates/leon/src/error.rs

```diff
@@ -0,0 +1,34 @@
+use miette::{Diagnostic, SourceSpan};
+use thiserror::Error;
+
+#[derive(Debug, Diagnostic, Error)]
+pub enum LeonError {
+    /// The template failed to parse.
+    #[error(transparent)]
+    InvalidTemplate(
+        #[diagnostic_source]
+        #[from]
+        ParseError,
```

Comment:
Nit: `ParseError` is quite large: `size_of::<String>()` is 24, `size_of::<SourceSpan>()` is 16 as it contains two `SourceOffset` which contains `usize`.

The `size_of::<ParseError>()` is probably `24 + (16 + 8 /* discriminant for Option */) * 2 = 24 * 3 = 72` bytes large, I think it makes sense to `Box` it here.

Reply:
Let's leave this kind of optimisation until the design is done. When things get CoWed it probably will change